packages/jest-environment/src/index.ts

```diff
@@ -60,6 +60,17 @@ export interface Jest {
    * executed within this time frame will be executed.
    */
   advanceTimersByTime(msToRun: number): void;
+  /**
+   * Advances all timers by `msToRun` milliseconds, firing callbacks if necessary.
+   *
+   * Also breaks the event loop, allowing any scheduled promise callbacks to execute _before_ running the timers.
+   *
+   * @returns
+   * Fake milliseconds since the unix epoch.
+   * @remarks
+   * Not available when using legacy fake timers implementation.
+   */
+  advanceTimersByTimeAsync(msToRun: number): Promise<number>;
```

Comment:
Why did you decide to return a value? That looks unexpected. If sync methods do not return, I think should be similar. Or both should return. But is the returned value useful? Nobody missed it in sync methods, so perhaps that is just a clutter? Or I missed something?

Reply:
Agreed - this should be the same across both implementations, and _if_ we wanna return something, that should be in a separate PR (for both sync and async, and probably the legacy implementation)